### PR TITLE
feat: TransactionService changes for expenses POST to platform

### PR DESCRIPTION
**Pull request by arjunaj5**
## Clickup
https://app.clickup.com/t/86cwgw5af

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 1]
```diff
@@ -342,8 +342,8 @@ export class ExpensesService {
     return expense;
   }
 
-  post(expense: Partial<Expense>): Observable<void> {
-    return this.spenderService.post<void>('/expenses', {
+  post(expense: Partial<Expense>): Observable<PlatformApiResponse<Expense>> {
+    return this.spenderService.post<PlatformApiResponse<Expense>>('/expenses', {
```
Comment (arjunaj5) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:346]: post /expenses returns the expense

File: src/app/core/services/transactions-outbox.service.ts [hunk 2]
```diff
@@ -405,7 +405,7 @@ export class TransactionsOutboxService {
 
             isInstafyleEnabled$.subscribe((isInstafyleEnabled) => {
               if (isInstafyleEnabled) {
-                that.addDataExtractionEntry(resp, entry.dataUrls);
+                // that.addDataExtractionEntry(resp, entry.dataUrls);
```
Comment (arjunaj5) [inline src/app/core/services/transactions-outbox.service.ts:408]: commenting out for now, this line of code will be removed in future PRs

File: src/app/shared/components/capture-receipt/capture-receipt.component.ts [hunk 3]
```diff
@@ -118,7 +118,6 @@ export class CaptureReceiptComponent implements OnInit, OnDestroy, AfterViewInit
         }
         const transaction = {
           source,
-          currency: eou?.org?.currency,
```
Comment (arjunaj5) [inline src/app/shared/components/capture-receipt/capture-receipt.component.ts:121]: currency not required while creating expense from receipt file_id

File: src/app/core/services/transaction.service.ts [hunk 4]
```diff
@@ -192,14 +193,19 @@ export class TransactionService {
     txn: Partial<Transaction>,
     fileUploads$: Observable<FileObject[]>
   ): Observable<Partial<Transaction>> {
-    const upsertTxn$ = this.upsert(txn);
-    return forkJoin([fileUploads$, upsertTxn$]).pipe(
-      switchMap(([fileObjs, transaction]) => {
-        const fileIds = fileObjs.map((fileObj) => fileObj.id);
-        if (fileIds.length > 0) {
-          return this.expensesService.attachReceiptsToExpense(transaction.id, fileIds).pipe(map(() => transaction));
```
Comment (arjunaj5) [inline src/app/core/services/transaction.service.ts:200]: removing this API call, instead we will pass the file_ids in the POST of /expenses itself.

File: src/app/core/models/image.model.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export interface Image {
+  source?: string;
+  base64Image?: string;
+}
```
Comment (arjunaj5) [inline src/app/core/models/image.model.ts:4]: fixing lint error

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 6]
```diff
@@ -342,14 +342,14 @@ export class ExpensesService {
...
+  post(expense: Partial<Expense>): Observable<{ data: Expense }> {
+    return this.spenderService.post<{ data: Expense }>('/expenses', {
       data: expense,
     });
   }
 
-  createFromFile(fileId: string, source: string): Observable<PlatformApiResponse<Expense[]>> {
-    return this.spenderService.post<PlatformApiResponse<Expense[]>>('/expenses/create_from_file/bulk', {
+  createFromFile(fileId: string, source: string): Observable<{ data: Expense[] }> {
+    return this.spenderService.post<{ data: Expense[] }>('/expenses/create_from_file/bulk', {
```
Comment (arjunaj5) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:352]: return types was incorrect before

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**